### PR TITLE
Preload webcomponents.js to prevent race with jQuery

### LINKED ISSUE #17: Some jQuery functions do not work when loading the Chromecast framework on iOS Chrome
If the Chromecast framework and jQuery are included on the same page, there is a race between the initialization of jQuery and `webcomponents.js` (which is loaded by the Chromecast framework when the browser does not have `document.registerElement`, e.g. Chrome on iOS). [A demo of this can be seen here.](https://cdn.rawgit.com/onebytegone/941cb2a3128b474a63908b86b522dff6/raw/03cb09ea8b27dba29eeef902ce8dcdc36dcc7afb/jquery-webcomponents-conflict.html) If the demo is opened in Safari on iOS, the blue box will animate up and down (using `jQuery.animate()`). However if the demo is opened on Chrome on iOS, the box will stay at the bottom of its container and not move.

[box-that-should-move](https://cdn.rawgit.com/onebytegone/941cb2a3128b474a63908b86b522dff6/raw/03cb09ea8b27dba29eeef902ce8dcdc36dcc7afb/jquery-webcomponents-conflict.html)

When the box fails to move, it is due an error thrown by the [`assertIsNodeWrapper` assertion](https://github.com/webcomponents/webcomponentsjs/blob/v0.7.24/webcomponents.js#L1609) that `webcomponents.js` added to [`insertBefore`](https://github.com/webcomponents/webcomponentsjs/blob/v0.7.24/webcomponents.js#L1822-L1865). That assertion will fail when jQuery wins the initialization race against `webcomponents.js`. On load, [jQuery creates a `div`](https://github.com/jquery/jquery/blob/2.1.4/src/css/support.js#L9) to use later [for pixel position calculations](https://github.com/jquery/jquery/blob/2.1.4/src/css/support.js#L28-L43). When `webcomponents.js` is loaded, [it makes a number of changes to `window.Node`](https://github.com/webcomponents/webcomponentsjs/blob/v0.7.24/webcomponents.js#L1817-L2065). Therefore if jQuery is initialized before `webcomponents.js`, the `div` jQuery created is of a different prototype than what `assertIsNodeWrapper` in `webcomponents.js` is expecting. However if  `webcomponents.js` initializes before jQuery, both libraries look to play well together.

While this issue is not directly caused by this plugin (or Video.js), for this plug-in to be the most versatile, it seems we should provide an easy work-around for the problem. This plugin should be loaded before the Chromecast framework and does not depend on jQuery, therefore this provides an opportunity to "preload" `webcomponents.js`. 

## Effected browsers

Chrome on iOS 9+

### PR DESCRIPTION
**Pull request by onebytegone**
This is to fix #17. The below command can be used to build a test case to verify this branch will actually fix the issue.

```bash
git clone git@github.com:silvermine/videojs-chromecast.git silvermine-videojs-chromecast
cd silvermine-videojs-chromecast
git checkout preload_webcomponents
curl -s https://gist.githubusercontent.com/onebytegone/941cb2a3128b474a63908b86b522dff6/raw/6bf15ce0eb41cc5a1629c353e8fba9b98e07cd98/jquery-webcomponents-conflict.html > docs/demo/jquery-webcomponents-conflict.html
npm install
grunt build
http-server -p 8080 -c-1
# Open http://IP_ADDRESS_HERE:8080/docs/demo/jquery-webcomponents-conflict.html in Chrome on iOS
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -22,6 +22,15 @@ process directly, or you can install the plugin as an npm module, include the
 javascript and SCSS source in your project using a Common-JS module loader and SASS build
 process, and copy the images from the image source folder to your project.
 
+Note that regardless of whether you are using this plugin via the pre-built JS or as a
+module, the Chromecast framework will need to be included after the plugin. For example:
+
+```html
+<script src="https://unpkg.com/video.js@6.1.0/dist/video.js"></script>
+<script src="../../dist/silvermine-videojs-chromecast.min.js"></script>
```
Comment (jthomerson) [inline README.md]: Why the `../..`? Seems like this is copy/paste from somewhere. Maybe just replace `../..` with `.`?

File: README.md [hunk 2]
```diff
@@ -42,6 +51,39 @@ Note: when adding the plugin's javascript to your web page, include the
 Video.js. The plugin's built javascript file expects there to be a reference to Video.js
 at `window.videojs` and will throw an error if it does not exist.
 
+### Initialization options
+
+* **`preloadWebComponents`** (default: `false`) - The Chromecast framework relies on the
+  `webcomponents.js` polyfill when a browser does not have `document.registerElement`.
+  If you are using jQuery, this polyfill must be loaded and initialized before jQuery is
+  initialized. Unfortunately, the Chromecast frameworks loads the `webcomponents.js`
```
Comment (jthomerson) [inline README.md]: `s/frameworks/framework/`

File: README.md [hunk 3]
```diff
@@ -42,6 +51,39 @@ Note: when adding the plugin's javascript to your web page, include the
...
+  polyfill via a dynamically created `<script>` tag. This causes a race condition (see
+  #17). Setting `preloadWebComponents` to `true` will make this plugin add the
+  `webcomponents.js` polyfill synchronously when the polyfill is needed.
```
Comment (jthomerson) [inline README.md]: Should add a bit more to this paragraph to explain what that means. For example:

> If you use the `preloadWebComponents: true` option, you should make sure that this plugin is loaded before jQuery. Then include the framework after this plugin as you normally would.
>
> tl;dr: if you use jQuery, you should use the `preloadWebComponents: true` option in this plugin.

File: src/js/preloadWebComponents.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+       androidChrome;
+
+   androidChrome = doesUserAgentContainString('Android')
+      && doesUserAgentContainString('Chrome/')
+      && window.navigator.presentation;
+
+   // These check are based on the checks found in `cast_sender.js` which
+   // determine if `cast_framework.js` needs to be loaded
+   if ((androidChrome || iosChrome) && needsWebComponents) {
+      require('webcomponents.js'); // eslint-disable-line global-require
```
Comment (jthomerson) [inline src/js/preloadWebComponents.js]: Let's add a little bit more commenting here. These two things come to mind:

 1. See `https://github.com/silvermine/videojs-chromecast/issues/17`
 2. We use webcomponents.js@0.7.24 because that's what was used by the Chromecast framework at the time this was added.

File: src/js/index.js [hunk 5]
```diff
@@ -14,11 +16,19 @@ var createChromecastButton = require('./components/ChromecastButton'),
  * {@link module:ChromecastButton} and {@link module:enableChromecast} for more details
  * about how the plugin and button are registered and configured.
  *
- * @param {object} videojs
+ * @param videojs {object} the videojs library. If `undefinded`, this plugin
```
Comment (jthomerson) [inline src/js/index.js]: undefinded?

File: src/js/preloadWebComponents.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+module.exports = function() {
+   var needsWebComponents = !document.registerElement,
+       iosChrome = doesUserAgentContainString('CriOS'),
+       androidChrome;
+
+   androidChrome = doesUserAgentContainString('Android')
+      && doesUserAgentContainString('Chrome/')
+      && window.navigator.presentation;
+
+   // These check are based on the checks found in `cast_sender.js` which
```
Comment (jthomerson) [inline src/js/preloadWebComponents.js]: `s/check/checks/`

**Comment by onebytegone** [comment]
@jthomerson Requested changes have been made
